### PR TITLE
OSS release candidate

### PR DESCRIPTION
**Pull request by matthayter**

### CODE REVIEW

File: tsd/pom.xml [hunk 1]
```diff
@@ -28,7 +28,7 @@
   <packaging>pom</packaging>
   <name>Metrics Tsd Parent</name>
   <description />
-  <version>0.3.6</version>
+  <version>0.4.0.GRPN.78-SNAPSHOT</version>
```
Comment (matthayter) [inline tsd/pom.xml]: I'd wager this is wrong

File: tsd/cluster-aggregator/pom.xml [hunk 2]
```diff
@@ -18,7 +18,7 @@
   <parent>
     <groupId>com.arpnetworking.metrics</groupId>
     <artifactId>tsd-parent</artifactId>
-    <version>0.3.6</version>
+    <version>0.4.0.GRPN.78-SNAPSHOT</version>
```
Comment (matthayter) [inline tsd/cluster-aggregator/pom.xml]: Probs wrong

File: tsd/tsd-core/pom.xml [hunk 3]
```diff
@@ -18,7 +18,7 @@
   <parent>
     <groupId>com.arpnetworking.metrics</groupId>
     <artifactId>tsd-parent</artifactId>
-    <version>0.3.6</version>
+    <version>0.4.0.GRPN.78-SNAPSHOT</version>
```
Comment (matthayter) [inline tsd/tsd-core/pom.xml]: Probably wrong

**Comment by vjkoskela** [comment]
So I can't view the entire diff Github of 470 files. 

Regarding the version for tsd (and its sub-modules) it should be set to a version lower than the one used in MAD and CAgg projects. Not that this code will be released, but it should be clear that MAD and CAgg replace it. Obviously, you must also remove the GRPN version extension. 

Aside from that the changes I was able to view look good to me. 

Thanks for taking the time to do this!

**Comment by matthayter** [comment]
@vjkoskela I suspect you already know, but you can fetch this branch locally (on you laptop) and use git diff to view the full changes, if you want to.

I'm struggling to get my head around what the versions should be - would you or @BrandonArp be able to make a suggestion / push directly to this branch? Would it be silly to put `0.3.7` on there?

**Comment by vjkoskela** [comment]
@matthayter Yeah thanks. I was not on a computer so that was not possible (afaik) on my iPhone.

The version of MAD and CAgg started at 1.0.0 so 0.3.7 or even what you have minus GRPN would be fine.

The diff looks good. 

One question, I don't think the files _build/start.sh_ and _build/pinger.sh_ should be deleted. Best I recall this had already been adapted to work across multiple repositories when launching the full system. Can you take a look?

**Comment by matthayter** [comment]
Ah yep, looks like this line in the `oss-clean.sh` excludes copying the build dir:
```
safe_command "rsync -r --exclude \".git/\" --exclude \".metadata/\" --exclude \".idea/\" --exclude \"target/\" --exclude \"build/\" --exclude \"logs/\" ${dir}/ /var/tmp/metrics.checks/"
```
So you won't get any changes to the `build` dir from an OSS release. I'll update the 'delete files' commit to not touch anything in build, as it looks like the GRPN metrics repo didn't delete anything in there either.

**Comment by matthayter** [comment]
@vjkoskela okey all done

**Comment by vjkoskela** [comment]
@matthayter I don't see the changes I was expecting to start.sh; although it does not appear to be getting deleted now which is a bonus ;)

Are there any open pull requests against the metrics repository on your end that were not merged?

If not, I guess those changes are lost. Is this PR good to go otherwise? (the title still reads do not merge)

**Comment by matthayter** [comment]
Oookey, I added the `build/` dir changes by hand as the `oss_clean` script excludes them. NOW we should be good to go; I'll remove the [Do not merge]; merge at will.

**Comment by vjkoskela** [comment]
So it looks like you didn't scrub some of the files. I don't know if I found them all, these were the obvious ones like grpn, groupon, snc1, and lup1. There may be others that I missed.

build/start/clusteragg/alerts/mad_dev.mad
+            "notify": "aint-dev@groupon.com",
^ should be some fake email (e.g. alert@example.com)

build/start.sh
+dir_mad=`find_path_up metrics-aggregator-daemon-grpn`
^ should be public repository name (e.g. metrics-aggregator-daemon)
^^ technically the same for metrics-remet-gui; although this at least doesn't say Groupon :)

build/start/clusteragg/clusterPipeline.json
+                                        "uri": "http://kmond-app-uat-vip.snc1:80/results",
^ should not be an snc1 address
^^ same in: build/start/clusteragg/hostPipeline.json

tsd/tsd-core/src/test/java/com/arpnetworking/tsdcore/sinks/DomainInjecting
SinkTest.java
+                .setSuffixes(ImmutableSet.of("lup1"))
^ should not be a lup1 domain

I'm okay merging it as-is; I'll leave it up to you @matthayter what you want to do. Just let me know!

**Comment by matthayter** [comment]
I made the following changes and then rebased to remove the history of the grpn stuff: https://gist.github.com/matthayter/f8812baf4bee120db467b9d7d8e19498

**Comment by vjkoskela** [comment]
Thanks @matthayter! Merged.

**Pull request merged**